### ISSUE TITLE
I found a crash when Android ui  automation, about Accessibility.

### ISSUE THREAD

**Issue by wangbo4020** [flutter/flutter#55210]
**I found a crash when Android ui  automation**

```
04-20 14:47:45.048 14860 14860 E AndroidRuntime: java.lang.NullPointerException: Attempt to read from field 'int android.graphics.Rect.left' on a null object reference
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at io.flutter.view.AccessibilityBridge.createAccessibilityNodeInfo(AccessibilityBridge.java:634)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchDescendantsOfVirtualNode(AccessibilityInteractionController.java:1264)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchDescendantsOfRealNode(AccessibilityInteractionController.java:1177)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchDescendantsOfRealNode(AccessibilityInteractionController.java:1173)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchDescendantsOfRealNode(AccessibilityInteractionController.java:1173)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchDescendantsOfRealNode(AccessibilityInteractionController.java:1173)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchAccessibilityNodeInfos(AccessibilityInteractionController.java:975)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController.findAccessibilityNodeInfoByAccessibilityIdUiThread(AccessibilityInteractionController.java:336)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController.access$400(AccessibilityInteractionController.java:67)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.view.AccessibilityInteractionController$PrivateHandler.handleMessage(AccessibilityInteractionController.java:1327)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:224)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7238)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
04-20 14:47:45.048 14860 14860 E AndroidRuntime: 	at 
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:876)
```

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E287,
    locale zh-Hans-CN)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
[✓] Android Studio (version 3.6)
[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.
```

**Comment by VladyslavBondarenko** [comment]
Hi @wangbo4020 
could you provide steps to reproduce it?

**Comment by wangbo4020** [comment]
> Hi @wangbo4020
> could you provide steps to reproduce it?

I have a more detailed log

**Comment by wangbo4020** [comment]
time in crash 04-18 18:07:41.595

[logcat.log](https://github.com/flutter/flutter/files/4508514/logcat.log)

**Comment by wangbo4020** [comment]
```
04-20 14:48:21.477   605  7789 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:21.527  3356  3633 D Settings/BW: ReadCodeTask:line content: 16
04-20 14:48:21.527  3356  3633 W ContextImpl: Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:765 android.content.ContextWrapper.sendBroadcast:396 com.vivo.daemonService.batterywarning.ReadCodeTask.sendWarningMessage:63 com.vivo.daemonService.batterywarning.ReadCodeTask.run:42 java.util.Timer$TimerImpl.run:284 
04-20 14:48:21.527  3356  3633 D Settings/BW: ReadCodeTask:notify code is 16
04-20 14:48:21.527  3356  3356 I WarningMessage: receiver: action ismediatek.intent.action.SAFETY_TIMER_TIMEOUT mBatteryPowerConnetDialog=true mShowChargerOverVoltageDialog=true
04-20 14:48:21.527  3356  3356 I WarningMessage: receiver: over 12 hours, battery does not charge full, please disconnect charger
04-20 14:48:21.527  3356  3633 D Settings/BW: ReadCodeTask:sendBroadcast: mediatek.intent.action.SAFETY_TIMER_TIMEOUT
04-20 14:48:21.547  2169  2169 E flutter : [ERROR:flutter/shell/platform/android/platform_view_android_jni.cc(39)] java.lang.NullPointerException: Attempt to read from field 'int android.graphics.Rect.left' on a null object reference
04-20 14:48:21.547  2169  2169 E flutter : 	at io.flutter.view.AccessibilityBridge.createAccessibilityNodeInfo(AccessibilityBridge.java:634)
04-20 14:48:21.547  2169  2169 E flutter : 	at android.view.accessibility.AccessibilityRecord.setSource(AccessibilityRecord.java:149)
04-20 14:48:21.547  2169  2169 E flutter : 	at io.flutter.view.AccessibilityBridge.obtainAccessibilityEvent(AccessibilityBridge.java:1559)
04-20 14:48:21.547  2169  2169 E flutter : 	at io.flutter.view.AccessibilityBridge.sendWindowContentChangeEvent(AccessibilityBridge.java:1535)
04-20 14:48:21.547  2169  2169 E flutter : 	at io.flutter.view.AccessibilityBridge.updateSemantics(AccessibilityBridge.java:1312)
04-20 14:48:21.547  2169  2169 E flutter : 	at io.flutter.view.AccessibilityBridge$1.updateSemantics(AccessibilityBridge.java:278)
04-20 14:48:21.547  2169  2169 E flutter : 	at io.flutter.embedding.engine.FlutterJNI.updateSemantics(FlutterJNI.java:438)
04-20 14:48:21.547  2169  2169 E flutter : 	at android.os.MessageQueue.nativePollOnce(Native Method)
04-20 14:48:21.547  2169  2169 E flutter : 	at android.os.MessageQueue.next(MessageQueue.java:323)
04-20 14:48:21.547  2169  2169 E flutter : 	at android.os.Looper.loop(Looper.java:135)
04-20 14:48:21.547  2169  2169 E flutter : 	at android.app.ActivityThread.main(ActivityThread.java:5665)
04-20 14:48:21.547  2169  2169 E flutter : 	at java.lang.reflect.Method.invoke(Native Method)
04-20 14:48:21.547  2169  2169 E flutter : 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:845)
04-20 14:48:21.547  2169  2169 E flutter : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:735)
04-20 14:48:21.547  2169  2169 E flutter : 
04-20 14:48:21.547  2169  2169 F flutter : [FATAL:flutter/shell/platform/android/platform_view_android_jni.cc(95)] Check failed: CheckException(env). 
04-20 14:48:21.547  2169  2169 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 2169 (droid.com.aaa)
04-20 14:48:21.607   581   581 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
04-20 14:48:21.607   581   581 F DEBUG   : Build fingerprint: 'vivo/PD1619/PD1619:6.0.1/MMB29M/compiler06292144:user/release-keys'
04-20 14:48:21.607   581   581 F DEBUG   : Revision: '0'
04-20 14:48:21.607   581   581 F DEBUG   : ABI: 'arm64'
04-20 14:48:21.607   581   581 F DEBUG   : pid: 2169, tid: 2169, name: droid.com.aaa  >>> client.android.com.aaa <<<
04-20 14:48:21.607   581   581 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
04-20 14:48:21.627   605  7789 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:21.627   605  7789 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:21.627   605  7789 D vivo_daemon:VvDmSrv: ********END
04-20 14:48:21.637   605  7789 D vivo_daemon:VvDmSrv: normal termination, exit status = 1
04-20 14:48:21.637  5795  5999 V RecordThread: SCREEN_RECORD end
04-20 14:48:21.637  5795  5999 V RecordThread: SCREEN_RECORD begin
04-20 14:48:21.637   605  7814 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:21.667   581   581 F DEBUG   : Abort message: '[FATAL:flutter/shell/platform/android/platform_view_android_jni.cc(95)] Check failed: CheckException(env). 
04-20 14:48:21.667   581   581 F DEBUG   : '
04-20 14:48:21.667   581   581 F DEBUG   :     x0   0000000000000000  x1   0000000000000879  x2   0000000000000006  x3   0000000000000000
04-20 14:48:21.667   581   581 F DEBUG   :     x4   0000000000000000  x5   0000000000000001  x6   0000000000000000  x7   0000000000000000
04-20 14:48:21.667   581   581 F DEBUG   :     x8   0000000000000083  x9   00000000000270d0  x10  000000002cee6fb0  x11  000000002cee7020
04-20 14:48:21.667   581   581 F DEBUG   :     x12  0000000000000000  x13  0000007f8511f000  x14  0000000000000000  x15  0000007f85111000
04-20 14:48:21.667   581   581 F DEBUG   :     x16  0000007f85111568  x17  0000007f850a42cc  x18  0000007f85122f50  x19  0000007f8552a088
04-20 14:48:21.667   581   581 F DEBUG   :     x20  0000007f85529fc8  x21  000000000000000b  x22  0000000000000006  x23  0000000000000000
04-20 14:48:21.667   581   581 F DEBUG   :     x24  0000000000000000  x25  00000000ffffffff  x26  000000000000000c  x27  0000000000000018
04-20 14:48:21.667   581   581 F DEBUG   :     x28  0000000000000018  x29  0000007fefb42590  x30  0000007f850a1a68
04-20 14:48:21.667   581   581 F DEBUG   :     sp   0000007fefb42590  pc   0000007f850a42d4  pstate 0000000020000000
04-20 14:48:21.677   581   581 F DEBUG   : 
04-20 14:48:21.677   581   581 F DEBUG   : backtrace:
04-20 14:48:21.677   581   581 F DEBUG   :     #00 pc 00000000000692d4  /system/lib64/libc.so (tgkill+8)
04-20 14:48:21.677   581   581 F DEBUG   :     #01 pc 0000000000066a64  /system/lib64/libc.so (pthread_kill+68)
04-20 14:48:21.677   581   581 F DEBUG   :     #02 pc 0000000000023728  /system/lib64/libc.so (raise+28)
04-20 14:48:21.677   581   581 F DEBUG   :     #03 pc 000000000001e058  /system/lib64/libc.so (abort+60)
04-20 14:48:21.677   581   581 F DEBUG   :     #04 pc 000000000028e924  /data/app/client.android.com.aaa-1/lib/arm64/libflutter.so (offset 0x270000)
04-20 14:48:21.677   581   581 F DEBUG   :     #05 pc 000000000028281c  /data/app/client.android.com.aaa-1/lib/arm64/libflutter.so (offset 0x270000)
04-20 14:48:21.677   581   581 F DEBUG   :     #06 pc 00000000002bd9e4  /data/app/client.android.com.aaa-1/lib/arm64/libflutter.so (offset 0x270000)
04-20 14:48:21.677   581   581 F DEBUG   :     #07 pc 0000000000290000  /data/app/client.android.com.aaa-1/lib/arm64/libflutter.so (offset 0x270000)
04-20 14:48:21.677   581   581 F DEBUG   :     #08 pc 00000000002926ec  /data/app/client.android.com.aaa-1/lib/arm64/libflutter.so (offset 0x270000)
04-20 14:48:21.677   581   581 F DEBUG   :     #09 pc 000000000001c004  /system/lib64/libutils.so (_ZN7android6Looper9pollInnerEi+524)
04-20 14:48:21.677   581   581 F DEBUG   :     #10 pc 000000000001c268  /system/lib64/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+80)
04-20 14:48:21.677   581   581 F DEBUG   :     #11 pc 00000000000d6990  /system/lib64/libandroid_runtime.so (_ZN7android18NativeMessageQueue8pollOnceEP7_JNIEnvP8_jobjecti+48)
04-20 14:48:21.677   581   581 F DEBUG   :     #12 pc 000000000237e87c  /system/framework/arm64/boot.oat (offset 0x237e000)
04-20 14:48:21.777   605  7814 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:21.777   605  7814 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:21.777   605  7814 D vivo_daemon:VvDmSrv: ********END
04-20 14:48:21.777   605  7814 D vivo_daemon:VvDmSrv: normal termination, exit status = 1
04-20 14:48:21.777  5795  5999 V RecordThread: SCREEN_RECORD end
04-20 14:48:21.777  5795  5999 V RecordThread: SCREEN_RECORD begin
04-20 14:48:21.777   605  6499 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:21.837  3586  4500 D DataFormatter: Special bytes : 0
04-20 14:48:21.837   597   678 I ThermalEngine: cpu_frequency_pre_req: SS-CPU5_0: device=cluster1 frequency=1958400
04-20 14:48:21.847   597   678 I ThermalEngine: cpu_frequency_pre_req: SS-CPU5_0: device=cluster1 frequency=1958400
04-20 14:48:21.857   462   462 I SurfaceFlinger: FPS: 53
04-20 14:48:21.877   597   678 I ThermalEngine: cpu_frequency_pre_req: SS-CPU5_0: device=cluster1 frequency=1958400
04-20 14:48:21.877   605  6499 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:21.877   605  6499 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:21.877   605  6499 D vivo_daemon:VvDmSrv: ********END
04-20 14:48:21.877   605  6499 D vivo_daemon:VvDmSrv: normal termination, exit status = 1
04-20 14:48:21.877  5795  5999 V RecordThread: SCREEN_RECORD end
04-20 14:48:21.877  5795  5999 V RecordThread: SCREEN_RECORD begin
04-20 14:48:21.877   605  3690 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:21.997   605  3690 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:21.997   605  3690 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:21.997   605  3690 D vivo_daemon:VvDmSrv: ********END
04-20 14:48:21.997   605  3690 D vivo_daemon:VvDmSrv: normal termination, exit status = 1
04-20 14:48:21.997  5795  5999 V RecordThread: SCREEN_RECORD end
04-20 14:48:21.997  5795  5999 V RecordThread: SCREEN_RECORD begin
04-20 14:48:21.997   605 19424 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:22.117   605 19424 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:22.117   605 19424 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:22.117   605 19424 D vivo_daemon:VvDmSrv: ********END
04-20 14:48:22.117   605 19424 D vivo_daemon:VvDmSrv: normal termination, exit status = 1
04-20 14:48:22.117  5795  5999 V RecordThread: SCREEN_RECORD end
04-20 14:48:22.117  5795  5999 V RecordThread: SCREEN_RECORD begin
04-20 14:48:22.117   605 32762 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:22.227   605 32762 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:22.227   605 32762 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:22.227   605 32762 D vivo_daemon:VvDmSrv: ********END
04-20 14:48:22.227   605 32762 D vivo_daemon:VvDmSrv: normal termination, exit status = 1
04-20 14:48:22.227  5795  5999 V RecordThread: SCREEN_RECORD end
04-20 14:48:22.227  5795  5999 V RecordThread: SCREEN_RECORD begin
04-20 14:48:22.227   605   605 W vivo_daemon:VvDmSrv: WARNING - runShellWithResult can return up to 2048 bytes, the rest are trimmed
04-20 14:48:22.337   605   605 D vivo_daemon:VvDmSrv: ********START, [1000:5795] showing 2048 Bytes maximum, v2.1
04-20 14:48:22.337   605   605 D vivo_daemon:VvDmSrv: root@android:/# screenrecord --time-limit 10 /sdcard/epm/now.mp4
04-20 14:48:22.337   605   605 D vivo_daemon:VvDmSrv: ********END
```

**Comment by tianpeng0703** [comment]
The same issue happened to me when I was running the monkey, have you found the solution?

**Comment by w4mxl** [comment]
Frequent monitoring of this exception, is there a solution?  @VladyslavBondarenko

**Comment by VladyslavBondarenko** [comment]
@w4mxl I don't know the solution, I'm doing only preliminary issue triage.
How can you reproduce it?

**Comment by w4mxl** [comment]
> @w4mxl I don't know the solution, I'm doing only preliminary issue triage.
> How can you reproduce it?

Sorry, I don't know how to reproduce it.We just saw this error log many times in the background of exception monitoring.

**Comment by chinmaygarde** [comment]
None of the stack frames are Flutter related. Closing because without a reproducible test case, it is unlikely that this is actionable.

**Comment by gaaclarke** [comment]
@chinmaygarde, @nscrobie is researching this issue.

**Comment by nscobie** [comment]
**TL;DR see related pending PR [#17656](https://github.com/flutter/engine/pull/17656) which appears to be about the same underlying issue.**

Hi all!

This has been a difficult issue to debug without a repro case, especially since I am new to the codebase. I have been digging through the code path throwing the NPE described by the logs wangbo4020 posted, and while I don't have a definitive solution, I hope this can provide some more context.

**Linked relevant trace (for branch [v1.12.13-hotfixes](https://github.com/flutter/engine/tree/v1.12.13-hotfixes))**

*Case 2 from logcat2.log posted by wangbo4020*

> 04-20 14:48:21.547  2169  2169 E flutter : [ERROR:flutter/shell/platform/android/platform_view_android_jni.cc(39)] java.lang.NullPointerException: Attempt to read from field 'int android.graphics.Rect.left' on a null object reference\
> 04-20 14:48:21.547  2169  2169 E flutter : at io.flutter.view.AccessibilityBridge.createAccessibilityNodeInfo([AccessibilityBridge.java:634](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L634))\
> 04-20 14:48:21.547  2169  2169 E flutter : at android.view.accessibility.AccessibilityRecord.setSource([AccessibilityRecord.java:149](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L149))\
> 04-20 14:48:21.547  2169  2169 E flutter : at io.flutter.view.AccessibilityBridge.obtainAccessibilityEvent([AccessibilityBridge.java:1559](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#1559))\
> 04-20 14:48:21.547  2169  2169 E flutter : at io.flutter.view.AccessibilityBridge.sendWindowContentChangeEvent([AccessibilityBridge.java:1535](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1535))\
> 04-20 14:48:21.547  2169  2169 E flutter : at io.flutter.view.AccessibilityBridge.updateSemantics([AccessibilityBridge.java:1312](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1312))\
> 04-20 14:48:21.547  2169  2169 E flutter : at io.flutter.view.AccessibilityBridge$1.updateSemantics([AccessibilityBridge.java:278](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L278))\
> 04-20 14:48:21.547  2169  2169 E flutter : at io.flutter.embedding.engine.FlutterJNI.updateSemantics(FlutterJNI.java:438)\
> 04-20 14:48:21.547  2169  2169 E flutter : at android.os.MessageQueue.nativePollOnce(Native Method)\
> 04-20 14:48:21.547  2169  2169 E flutter : at android.os.MessageQueue.next(MessageQueue.java:323)\
> 04-20 14:48:21.547  2169  2169 E flutter : at android.os.Looper.loop(Looper.java:135)\
> 04-20 14:48:21.547  2169  2169 E flutter : at android.app.ActivityThread.main(ActivityThread.java:5665)\
> 04-20 14:48:21.547  2169  2169 E flutter : at java.lang.reflect.Method.invoke(Native Method)\
> 04-20 14:48:21.547  2169  2169 E flutter : at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:845)\
> 04-20 14:48:21.547  2169  2169 E flutter : at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:735)

**Deep dive explanation:**

By following the trace above it looks like we are operating on the root virtual accessibility node/view, as [sendWindowContentChangeEvent(0);](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1312) is called in our failing code path and the 0 passed as a parameter corresponds to [ROOT_NODE_ID](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L75). In this version there was a warning when in debug mode when trying to call [obtainAccessbilityEvent(...)](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1553) with virtualViewId: 0, which would have fired in this case. There is a PR ([#14079](https://github.com/flutter/engine/pull/14079)) which has since removed that warning citing it as spurious, so I'm unsure if that would have been an issue in this version or not. Android then calls back into [createAccessibilityNodeInfo(...)](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L495), presumably with virtualViewId: 0. However, down near where the exception is being thrown it seems like this current node (again, presuming it's the root node) [has a parent which is not null](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L631), but that parent has not been properly initialized since its globalRect is null, which as @goderbauer [points out](https://github.com/flutter/engine/pull/17656#issuecomment-613039115) should probably never be the case. Each node's globalRect field should be initialized by calling [SemanticsNode.updateRecursively(...)](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L2098) when [SemanticsNode.globalGeometryDirty](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1832) == true, which are both handled by calling [AccessibilityBridge.updateSemantics(...)](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L1239) which I believe is ultimately called by the framework when the widget tree has been updated.

Again, I am not super familiar with the codebase yet, but my (admittedly vague) hunch is that we are making assumptions about the root accessibility node which don't always hold true for some manufacturers, either in the Java side of things (AccessibilityBridge and friends), the generalized C++ engine level above us (see [platform_view_android.cc](https://github.com/flutter/engine/blob/v1.12.13-hotfixes/shell/platform/android/platform_view_android.cc#L227)), or with the data we are being passed from the framework. More specifically, it feels to me like somehow an extraneous object is being parented above root (unsure how this base part of the tree works), or some similar shenanigans. Unsure why this would be happening on Huawei, Xiaomi, and Vivo devices specifically.

**Possible ways forward:**

1.  Is anyone who is experiencing this issue able to upgrade to or test with the latest stable version of Flutter? [PR #14079](https://github.com/flutter/engine/pull/14079) from December 2019 removed an apparently spurious warning which would have fired on the version you are running (v1.12.13+hotfix.9) when in debug mode. It is possible there were changes to other parts of the engine/framework which fixed an underlying issue prior to this. I don't have enough context to be sure, but it may be worth checking.
2.  **"Paper over" the issue by adding a null check. Other people seem to be experiencing this issue with Huawei/Xiaomi phones (on v1.9.1.hotfix.6), and an unmerged PR from April 2020 ([#17656](https://github.com/flutter/engine/pull/17656)) introduces a null check to fix the crash, but there are very valid concerns regarding hiding the underlying issue. If this is still an issue for customers I would suggest we follow up there. If a null check is introduced, we would need to carefully monitor for the underlying issue.**

/cc: @goderbauer @liyuqian

**Comment by wangbo4020** [comment]
On flutter v1.12.13+hotfix.9.
I upload to [vivo云测平台](https://vcl.vivo.com.cn/#/home/index), choice x9 or other vivo phones.
The test report fails, the log will be provided.

**Comment by gaaclarke** [comment]
Closing the issue since it appears to be fixed.  Unfortunately the fix hasn't been pinpointed exactly.

**Comment by jchenx2** [comment]
I upgrade flutter to 1.17.3,but this crash still exists
